rest-api/alembic/versions/6e5980478a0e_add_external_id.py

```diff
@@ -0,0 +1,64 @@
+"""add external id
+
+Revision ID: 6e5980478a0e
+Revises: 378ee128c23f
+Create Date: 2018-10-18 10:09:35.910876
+
+"""
+from alembic import op
+import sqlalchemy as sa
+import model.utils
+
+
+from participant_enums import PhysicalMeasurementsStatus, QuestionnaireStatus, OrderStatus
+from participant_enums import WithdrawalStatus, WithdrawalReason, SuspensionStatus, QuestionnaireDefinitionStatus
+from participant_enums import EnrollmentStatus, Race, SampleStatus, OrganizationType, BiobankOrderStatus
+from participant_enums import MetricSetType, MetricsKey
+from model.site_enums import SiteStatus, EnrollingStatus, DigitalSchedulingStatus, ObsoleteStatus
+from model.code import CodeType
+
+# revision identifiers, used by Alembic.
+revision = '6e5980478a0e'
+down_revision = '378ee128c23f'
+branch_labels = None
+depends_on = None
+
+
+def upgrade(engine_name):
+    globals()["upgrade_%s" % engine_name]()
+
+
+def downgrade(engine_name):
+    globals()["downgrade_%s" % engine_name]()
+
+
+
+def upgrade_rdr():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('participant', sa.Column('external_id', sa.BigInteger(), nullable=True))
+    op.create_unique_constraint(None, 'participant', ['external_id'])
+    op.add_column('participant_history', sa.Column('external_id', sa.BigInteger(), nullable=True))
+    op.create_unique_constraint(None, 'participant_history', ['external_id'])
```

Comment:
external_id can't be unique in history, as there will be multiple versions of the same participant... you'll need to add the unique constraint in participant only.

Reply:
🤦‍♂️ yep, fixed.